python_ta/debug/snapshot_manager.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+import inspect
+import logging
+import os
+import re
+import sys
+import types
+from typing import Any, Iterable, Optional
+
+from python_ta.debug.snapshot import snapshot
+
+
+class SnapshotManager:
```

Comment:
I'm not a big fan of the name "manager" here (naming indeed being a hard problem). Let's go with `SnapshotTracer`. Note that you should rename various files in addition to places in the code.